src/Microsoft.AspNetCore.Razor.Language/Legacy/CSharpCodeWriter.cs

```diff
@@ -651,5 +656,46 @@ public void Dispose()
             {
             }
         }
+
+        public struct StringLiteralWriter : IDisposable
```

Comment:
The PR is made to render a list of HTML token into the CSharpCodeWriter, but this list of token is divided into several "string literals", and these "string literals" are composed of the token contents (or even part of token contents).